app/jobs/sync_playbooks.rb

```diff
@@ -0,0 +1,31 @@
+module ImportPlaybooksJob
+  module Async
+    class SyncPlaybooks < ::Actions::EntryAction
+      require_relative '../services/foreman_ansible/playbooks_importer'
+
+      def plan(changed, proxy, playbooks_names)
+        plan_self(changed: changed, proxy: proxy, playbooks_names: playbooks_names)
+      end
+
+      def run
+        playbooks_importer = ForemanAnsible::PlaybooksImporter.new(proxy)
```

Comment:
There is an instance of importer in already created in the controller, so you can pass it in and re-use it. That way, you do not need a proxy and instantiate a new importer.